jep/227/compatibility.adoc

```diff
@@ -0,0 +1,320 @@
+looks like it should fail, in fact
+link:https://github.com/jenkinsci/cloudbees-folder-plugin/blob/24c66b5db8fcf9e6e67da4f07b6054d2ae1acf3f/src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java#L1048[this code]
+is throwing the Acegi Security version of the exception.
+At runtime, either should be accepted, due to `AcegiSecurityExceptionFilter` in core.
+
+|link:https://plugins.jenkins.io/coding-webhook/[coding-webhook]
+|Incompatible
+|Uses `CliAuthenticator`.
+
+|link:https://plugins.jenkins.io/copyartifact/[copyartifact]
+|Compatible
+|PCT passes.
+Has an unusual `AUTHENTICATED_ANONYMOUS` but believed safe.
+
+|link:https://plugins.jenkins.io/credentials/[credentials]
+|Compatible
+|PCT passes.
+Cleanups integrated as of link:https://github.com/jenkinsci/credentials-plugin/releases/tag/credentials-2.3.13[2.3.13].
+
+|link:https://plugins.jenkins.io/crowd/[crowd]
+|Incompatible
+|Uses `BeanBuilder` and possibly other problematic types.
+
+|link:https://plugins.jenkins.io/crowd2/[crowd2]
+|Incompatible
+|Uses some unsupported types from Acegi Security.
+
+|link:https://plugins.jenkins.io/declarative-pipeline-migration-assistant/[declarative-pipeline-migration-assistant]
+|Compatible
+|As of link:https://github.com/jenkinsci/declarative-pipeline-migration-assistant-plugin/releases/tag/declarative-pipeline-migration-assistant-1.1.2[1.1.2].
+
+|link:https://plugins.jenkins.io/dry-run/[dry-run]
+|Mostly compatible
+|link:https://github.com/jenkinsci/dry-run-plugin/blob/63733e771406943766a1ac6f9308701aca291309/src/main/java/org/jenkinsci/plugins/dryrun/DryRunProjectAction.java#L29-L34[this code]
+could be switched to `hasPermission` for full compatibility.
+
+|link:https://plugins.jenkins.io/extensible-choice-parameter/[extensible-choice-parameter]
+|Probably compatible
+|Unclear what the purpose of link:https://github.com/jenkinsci/extensible-choice-parameter-plugin/blob/3fda75de34d4099d078a038fc3eeb3c38679c1f8/src/test/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/SystemGroovyChoiceListProviderJenkinsTest.java#L217-L219[this test code] is.
+
+|link:https://plugins.jenkins.io/github-oauth/[github-oauth]
+|Incompatible, fixes prepared
+|link:https://github.com/jenkinsci/github-oauth-plugin/pull/119[github-oauth-plugin #119]
+is necessary.
+Given that, it seems to work.
+
+|link:https://plugins.jenkins.io/gitlab-oauth/[gitlab-oauth]
+|Unknown
+|Review and testing required.
+
+|link:https://plugins.jenkins.io/google-cloud-health-check/[google-cloud-health-check]
+|Probably compatible
+|link:https://github.com/jenkinsci/google-cloud-health-check-plugin/blob/a6b68119ebd64c1bad026ad5ed7a04c964bcb382/src/test/java/com/google/jenkins/plugins/health/HealthCheckActionTest.java#L63[This code]
+would need to be adjusted to make tests pass.
+
+|link:https://plugins.jenkins.io/google-login/[google-login]
+|Probably compatible
+|Untested.
+
+|link:https://plugins.jenkins.io/jenkins-cloudfoundry-uaa/[jenkins-cloudfoundry-uaa]
```

Comment:
This link doesn't work, the plugin doesn't exist. Is this https://plugins.jenkins.io/cloudfoundry/ ? Superficially perhaps not, couldn't find a `BeanBuilder` there at a glance.

Reply:
#308